include/RAJA/policy/cuda/teams.hpp

```diff
@@ -100,8 +97,74 @@ struct LaunchExecute<RAJA::expt::cuda_launch_t<async, 0>> {
       RAJA_FT_END;
     }
 
-    // return resources::EventProxy<resources::Cuda>(&cuda_res);
+    //resources::resource out_res(&cuda_res);
+    //return out_res;
+      //return resources::EventProxy<resources::Cuda>(&cuda_res);
+  }
+
+  template <typename BODY_IN>
+  static void exec(TeamResources &res, LaunchContext const &ctx, BODY_IN &&body_in)
+  {
+    using BODY = camp::decay<BODY_IN>;
+
+    auto func = launch_global_fcn<BODY>;
+
+    resources::Cuda &cuda_res = res.cuda; 
+      //resources::Cuda::get_default();
+    /* Use the zero stream until resource is better supported */
+    cudaStream_t stream = cuda_res.get_stream();
```

Comment:
@mdavis36 is this all we need? and does this look reasonable? Idea here is to pass in a TeamResource struct that holds all supported back-ends an we choose the right one depending on what is being called.

Reply:
If you want to support events that would need a bit more.  What's making the decision as to which backend to use? Is it controlled by the caller somehow?  I ask because if it is, it might make sense for it to be a variant-like type where the TeamResource only contains one resource, but could contain any of a compile-time list of them, if it's out of the caller's control then maybe tuple of them makes sense so the types are known.  If the types can't be known, then some more error checking is needed to make sure the resource is the right kind.

Reply:
Yes the caller chooses the backend, its base on run-time parameter in the launch API. Basically an enum for HOST or DEVICE.

Reply:
So, perhaps, pass a resource of type rather than enum to identify type could be a nice model?  If it's using the camp platform enum the number is already there in the erased resource type.

Reply:
That sounds pretty good to me, we had some other ideas as well. I'll explore them and push to this branch.